### PR TITLE
fix(driver/bpf): not sending all arguments on execve fail

### PR DESCRIPTION
**Pull request by mrgian**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

> /area driver-kmod

/area driver-bpf

/area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

> /area libsinsp

> /area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

> /version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:

When `execve` fails we get the arguments (or environment variables) from the systems call arguments instead of trying to get them from the kernel.
However, when one of the arguments is empty, the BPF driver doesn't send it to userspace, or even worse, doesn't send the following ones.
Example:
```
const char* argv[] = {"/non/existent", "arg0", "arg1", "", "arg3", NULL};
const char* envp[] = {"env0", "env1", "", "env3", NULL};
execve(argv[0], argv, envp);
```
Expected args: `arg0.arg1..arg3.`
Actual args (BPF): `arg0.arg1.arg3.`
Actual args (modern BPF): `arg0.arg1.` 

As you can see, in the case of BPF the empty argument is skipped.
In the case of modern BPF the empty argument and the following ones are skipped.

This PR fixes this issue.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
fix(driver/bpf): not sending entire arguments array on execve fail
```

### CODE REVIEW

File: driver/bpf/fillers.h [hunk 1]
```diff
@@ -1941,6 +1941,14 @@ static __always_inline int bpf_accumulate_argv_or_env(struct filler_data *data,
 		}
 
 		len = bpf_probe_read_user_str(&data->buf[off & SCRATCH_SIZE_HALF], SCRATCH_SIZE_HALF, arg);
```
Comment (Andreagit97) [inline driver/bpf/fillers.h:1943]: This is related to this https://github.com/falcosecurity/libs/issues/866, I would expect `bpf_probe_read_user_str` to return `1` when the string is empty but indeed it returns `0` not considering the NULL terminator... BTW your workaround makes sense

**Review by Andreagit97 — COMMENTED** [review]
Fixes are fine, thank you for this! Can we add a test for it in the `driver_tests` framework? Moreover, we need to check if the verifier accepts these fixes in the old probe because this code area is pretty delicate...

**Comment by mrgian** [comment]
Ei @Andreagit97 
> Can we add a test for it in the driver_tests framework? 

This issue came up while running one of the test @therealbobo is porting. It tests this specific case.
Maybe we can skip porting that test in `libsinsp_e2e` and adding a similar test in the `driver_tests` framework like you said. WDYT?

**Comment by Andreagit97** [comment]
> Maybe we can skip porting that test in libsinsp_e2e and adding a similar test in the driver_tests framework like you said. WDYT?

Yep, this seems the right way to go! in the end is not an e2e test but a simple driver test

**Comment by Andreagit97** [comment]
These are the results of the kernel testing against this branch:

ARM64

|        KERNEL        | CMAKE-CONFIGURE | KMOD BUILD | KMOD SCAP-OPEN |            BPF-PROBE BUILD             |            BPF-PROBE SCAP-OPEN             |            MODERN-BPF SCAP-OPEN             |
|----------------------|-----------------|------------|----------------|----------------------------------------|--------------------------------------------|---------------------------------------------|
| amazonlinux2-5.4     | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | [🟡](#amazonlinux2-54-modern-bpf_scap-open) |
| amazonlinux2022-5.15 | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                          |
| fedora-6.2           | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                          |
| oraclelinux-4.14     | 🟢              | 🟢         | 🟢             | [🟡](#oraclelinux-414-bpf-probe_build) | [🟡](#oraclelinux-414-bpf-probe_scap-open) | [🟡](#oraclelinux-414-modern-bpf_scap-open) |
| oraclelinux-5.15     | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                          |
| ubuntu-6.5           | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                          |

x86

|        KERNEL        | CMAKE-CONFIGURE | KMOD BUILD | KMOD SCAP-OPEN |            BPF-PROBE BUILD             |            BPF-PROBE SCAP-OPEN             |             MODERN-BPF SCAP-OPEN             |
|----------------------|-----------------|------------|----------------|----------------------------------------|--------------------------------------------|----------------------------------------------|
| amazonlinux2-4.19    | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | [🟡](#amazonlinux2-419-modern-bpf_scap-open) |
| amazonlinux2-5.10    | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| amazonlinux2-5.15    | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| amazonlinux2-5.4     | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | [🟡](#amazonlinux2-54-modern-bpf_scap-open)  |
| amazonlinux2022-5.15 | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| amazonlinux2023-6.1  | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| archlinux-6.0        | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| archlinux-6.7        | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| centos-3.10          | 🟢              | 🟢         | 🟢             | [🟡](#centos-310-bpf-probe_build)      | [🟡](#centos-310-bpf-probe_scap-open)      | [🟡](#centos-310-modern-bpf_scap-open)       |
| centos-4.18          | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| centos-5.14          | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| fedora-5.17          | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| fedora-5.8           | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| fedora-6.2           | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| oraclelinux-3.10     | 🟢              | 🟢         | 🟢             | [🟡](#oraclelinux-310-bpf-probe_build) | [🟡](#oraclelinux-310-bpf-probe_scap-open) | [🟡](#oraclelinux-310-modern-bpf_scap-open)  |
| oraclelinux-4.14     | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | [🟡](#oraclelinux-414-modern-bpf_scap-open)  |
| oraclelinux-5.15     | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |
| oraclelinux-5.4      | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | [🟡](#oraclelinux-54-modern-bpf_scap-open)   |
| ubuntu-4.15          | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | [🟡](#ubuntu-415-modern-bpf_scap-open)       |
| ubuntu-5.8           | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | [🟡](#ubuntu-58-modern-bpf_scap-open)        |
| ubuntu-6.5           | 🟢              | 🟢         | 🟢             | 🟢                                     | 🟢                                         | 🟢                                           |

So we should be fine

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**